### PR TITLE
Teams: Support team UIDs in APIs

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Support team UIDs in APIs

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by Jguer** [comment]
Do not merge until API level tests are implemented

File: pkg/services/accesscontrol/resourcepermissions/api.go [hunk 1]
```diff
@@ -42,21 +43,27 @@ func (a *api) registerEndpoints() {
 		licenseMW = nopMiddleware
 	}
 
+	teamUIDResolver := team.MiddlewareTeamUIDResolver(a.service.teamService, ":teamID")
+	teamUIDResolverResource := func() web.Handler { return func(c *contextmodel.ReqContext) {} }() // no-op
+	if a.service.options.Resource == "teams" {
+		teamUIDResolverResource = team.MiddlewareTeamUIDResolver(a.service.teamService, ":resourceID")
```
Comment (Jguer) [inline pkg/services/accesscontrol/resourcepermissions/api.go:49]: when porting extra resources like users to UIDs we should make this resolver an option

File: pkg/services/team/model.go [hunk 2]
```diff
@@ -54,8 +54,10 @@ type DeleteTeamCommand struct {
 }
 
 type GetTeamByIDQuery struct {
```
Comment (aarongodin) [inline pkg/services/team/model.go:56]: Maybe appropriate to change to `type GetTeamQuery`?

Similarly below, maybe it makes sense to change the method `GetTeamByID` on the service to just `GetTeam`?

File: public/app/types/teams.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export interface TeamDTO {
 // This is the team resource with permissions and metadata expanded
 export interface Team {
   id: number; // TODO switch to UUID
+  uid: string; // Prefer UUID
```
Comment (aarongodin) [inline public/app/types/teams.ts:18]: Is it really a UUID or a grafana short ID?

**Review by aarongodin — APPROVED** [review]
LGTM, although based on your comment I don't think now is a bad time to make the resolver generic to the resourcepermissions API instance. Fair if we want to tackle that later too.

File: public/app/types/teams.ts (hunk 3 above)
Comment (Jguer) [reply public/app/types/teams.ts:18]: Grafana short UID. can't really call it UUID :sweat:

File: pkg/services/accesscontrol/resourcepermissions/api.go [hunk 4]
```diff
@@ -42,21 +43,27 @@ func (a *api) registerEndpoints() {
 		licenseMW = nopMiddleware
 	}
 
+	teamUIDResolver := team.MiddlewareTeamUIDResolver(a.service.teamService, ":teamID")
```
Comment (kalleep) [inline pkg/services/accesscontrol/resourcepermissions/api.go:46]: is this one unused?

File: pkg/services/team/teamapi/api.go [hunk 5]
```diff
@@ -53,36 +57,58 @@ func ProvideTeamAPI(
 	return tapi
 }
 
+func TeamUIDResolver(teamService team.Service) web.Handler {
```
Comment (kalleep) [inline pkg/services/team/teamapi/api.go:60]: This can be replaces by `MiddlewareTeamUIDResolver` right?

Comment (Jguer) [reply pkg/services/team/teamapi/api.go:60]: This one is unused and I forgot to remove it, sorry

**Review by kalleep — APPROVED** [review]

**Pull request merged**